common/src/main/scala/services/teamwork/SquadSwitchboard.scala

```diff
@@ -0,0 +1,142 @@
+    * The message `SquadSwitchboard.Leave` removes the user from this collection.
+    * The message `SquadSwitchboard.Unwatch` also removes the user from this collection.
+    * key - unique character id; value - `Actor` reference for that character
+    */
+  val Watchers : mutable.LongMap[ActorRef] = mutable.LongMap[ActorRef]()
+
+  override def postStop() : Unit = {
+    UserActorMap.clear()
+    DelayedJoin.clear()
+    Watchers.clear()
+  }
+
+  def receive : Receive = {
+    case SquadSwitchboard.Join(char_id, Some(actor)) =>
+      UserActorMap(char_id) = DelayedJoin.remove(char_id).orElse( Watchers.remove(char_id)) match {
+        case Some(_actor) =>
+          context.watch(_actor)
+          _actor
+        case None =>
+          context.watch(actor)
+          actor
+      }
+
+    case SquadSwitchboard.Join(char_id, None) =>
+      DelayedJoin.remove(char_id).orElse( Watchers.remove(char_id)) match {
+        case Some(actor) =>
+          UserActorMap(char_id) = actor
+        case None => ;
+      }
+
+    case SquadSwitchboard.DelayJoin(char_id, actor) =>
+      context.watch(actor)
+      DelayedJoin(char_id) = actor
+
+    case SquadSwitchboard.Leave(char_id) =>
+      UserActorMap.find { case(charId, _) => charId == char_id }
+        .orElse(DelayedJoin.find { case(charId, _) => charId == char_id })
+        .orElse(Watchers.find { case(charId, _) => charId == char_id }) match {
+        case Some((member, actor)) =>
+          context.unwatch(actor)
+          UserActorMap.remove(member)
+          DelayedJoin.remove(member)
+          Watchers.remove(member)
+        case None => ;
+      }
+
+    case SquadSwitchboard.Watch(char_id, actor) =>
+      context.watch(actor)
+      Watchers(char_id) = actor
+
+    case SquadSwitchboard.Unwatch(char_id) =>
+      Watchers.remove(char_id)
+
+    case SquadSwitchboard.To(member, msg) =>
+      UserActorMap.find { case (char_id, _) => char_id == member } match {
+        case Some((_, actor)) =>
+          actor ! msg
+        case None => ;
+      }
+
+    case SquadSwitchboard.ToAll(msg) =>
+      UserActorMap
+        .foreach { case (_, actor) =>
+          actor ! msg
+        }
+
+    case SquadSwitchboard.Except(excluded, msg) =>
+      UserActorMap
+        .filterNot { case (char_id, _) => char_id == excluded }
+        .foreach { case (_, actor) =>
+          actor ! msg
+        }
+
+    case Terminated(actorRef) =>
+      UserActorMap.find { case(_, ref) => ref == actorRef }
+        .orElse(DelayedJoin.find { case(_, ref) => ref == actorRef })
+        .orElse(Watchers.find { case(_, ref) => ref == actorRef }) match {
+        case Some((member, actor)) =>
+          context.unwatch(actor)
+          UserActorMap.remove(member)
+          DelayedJoin.remove(member)
+          Watchers.remove(member)
+        case None => ;
+      }
+
+    case _ => ;
+  }
+}
+
+object SquadSwitchboard {
+  final case class Join(char_id : Long, actor : Option[ActorRef])
```

Comment:
Do we need to make these extend from a sealed trait like the other messages?

Reply:
We'll think about when `SquadSwitchboard` actually does something other than manage the periodic request-invite workaround.